tests/components/amberelectric/test_coordinator.py

```diff
@@ -0,0 +1,202 @@
+"""Tests for the Amber Electric Data Coordinator."""
+from typing import Generator
+from unittest.mock import Mock, patch
+
+from amberelectric import ApiException
+from amberelectric.model.channel import Channel, ChannelType
+from amberelectric.model.site import Site
+import pytest
+
+from homeassistant.components.amberelectric.coordinator import AmberUpdateCoordinator
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import UpdateFailed
+
+from tests.components.amberelectric.helpers import (
+    CONTROLLED_LOAD_CHANNEL,
+    FEED_IN_CHANNEL,
+    GENERAL_AND_CONTROLLED_SITE_ID,
+    GENERAL_AND_FEED_IN_SITE_ID,
+    GENERAL_CHANNEL,
+    GENERAL_ONLY_SITE_ID,
+)
+
+
+@pytest.fixture(name="current_price_api")
+def mock_api_current_price() -> Generator:
+    """Return an authentication error."""
+    instance = Mock()
+
+    general_site = Site(
+        GENERAL_ONLY_SITE_ID,
+        "11111111111",
+        [Channel(identifier="E1", type=ChannelType.GENERAL)],
+    )
+    general_and_controlled_load = Site(
+        GENERAL_AND_CONTROLLED_SITE_ID,
+        "11111111112",
+        [
+            Channel(identifier="E1", type=ChannelType.GENERAL),
+            Channel(identifier="E2", type=ChannelType.CONTROLLED_LOAD),
+        ],
+    )
+    general_and_feed_in = Site(
+        GENERAL_AND_FEED_IN_SITE_ID,
+        "11111111113",
+        [
+            Channel(identifier="E1", type=ChannelType.GENERAL),
+            Channel(identifier="E2", type=ChannelType.FEED_IN),
+        ],
+    )
+    instance.get_sites.return_value = [
+        general_site,
+        general_and_controlled_load,
+        general_and_feed_in,
+    ]
+
+    with patch("amberelectric.api.AmberApi.create", return_value=instance):
+        yield instance
+
+
+async def test_fetch_general_site(hass: HomeAssistant, current_price_api: Mock) -> None:
+    """Test fetching a site with only a general channel."""
+
+    current_price_api.get_current_price.return_value = GENERAL_CHANNEL
+    data_service = AmberUpdateCoordinator(hass, current_price_api, GENERAL_ONLY_SITE_ID)
```

Comment:
Please test the coordinator indirectly by testing the sensor state instead.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations